integration-test/src/main/java/uk/gov/gchq/gaffer/integration/junit/extensions/IntegrationTestSuiteExtension.java

```diff
@@ -0,0 +1,320 @@
+     * {@link ConditionEvaluationResult#disabled(String)} {@link Object} for the
+     * {@link org.junit.jupiter.api.Test} in question
+     */
+    @Override
+    public ConditionEvaluationResult evaluateExecutionCondition(final ExtensionContext context) {
+        if (context.getTestMethod().isPresent()) {
+            final String currentMethodName = context.getTestMethod().get().getName();
+            if (this.skipTestMethods.containsKey(currentMethodName)) {
+                return ConditionEvaluationResult.disabled(this.skipTestMethods.get(currentMethodName));
+            }
+        }
+        return ConditionEvaluationResult.enabled("Test enabled");
+    }
+
+    private static Set<Object> getSuiteObjects(final IntegrationTestSuite integrationTestSuite) {
+        final Set<Object> objects = integrationTestSuite.getObjects().orElse(Collections.emptySet());
+        LOGGER.debug("Retrieved the following objects types from the IntegrationTestSuite: [{}]",
+                objects.stream().map(o -> o.getClass().getName()).collect(Collectors.toList()));
+        return objects;
+    }
+
+    private Map<String, String> getSkipTestMethods(final IntegrationTestSuite integrationTestSuite) {
+        final Map<String, String> skipTestMethods = integrationTestSuite.getTestsToSkip().orElse(Collections.emptyMap());
+        LOGGER.debug("Retrieved the following skip-test methods from the IntegrationTestSuite: [{}]",
+                StringUtils.join(skipTestMethods));
+        return skipTestMethods;
+    }
+
+    private void injectInstanceFields(final Object instance) {
+        findAnnotatedFields(instance.getClass(), IntegrationTestSuiteInstance.class, ReflectionUtils::isNotStatic).forEach(field -> {
+            try {
+                LOGGER.debug("Field [{}] requires injecting", field);
+                final Object object = getObject(field.getType());
+                LOGGER.debug("Object [{}] found for the field", object);
+                makeAccessible(field).set(instance, object);
+            } catch (final Throwable t) {
+                ExceptionUtils.throwAsUncheckedException(t);
+            }
+        });
+    }
+
+    private Object getObject(final Class<?> type) {
+        final Set<Object> set = suiteCache.stream()
+                .filter(o -> type.isAssignableFrom(o.getClass()))
+                .collect(Collectors.toSet());
+        if (!set.isEmpty()) {
+            return set.iterator().next();
+        } else {
+            throw new ParameterResolutionException(String.format("Object of type [%s] not found", type));
+        }
+    }
+
+    private static IntegrationTestSuite getIntegrationTestSuite(final String initClass) {
+        final IntegrationTestSuite integrationTestSuite;
+        if (INTEGRATION_TEST_SUITE_CLASS_MAP.containsKey(initClass)) {
+            integrationTestSuite = INTEGRATION_TEST_SUITE_CLASS_MAP.get(initClass);
+        } else {
+            synchronized (INTEGRATION_TEST_SUITE_CLASS_MAP) {
```

Comment:
I thought you ran the modules in parallel to reduce the build time? As this is static would it be affected by this? Although, as the differing modules have differing classes this may be redundant.

Reply:
Only the compilation of modules can be run in parallel so far as I know (`-Pquick`). Running tests for the modules in parallel doesn't work - possibly due to the MiniAccumuloStore. We don't run any tests in parallel, if we did this might be needed - though I think you're likely right about the differing classes.

Reply:
I'll remove the `synchronized`